src/getMessageAuthor.ts

```diff
@@ -88,3 +119,40 @@ function authorFromGitHubContext(): MessageAuthor | null {
     icon_url: sender.avatar_url
   }
 }
+
+async function getSenderFromPRMerger(
+  octokit: OctokitClient
+): Promise<MessageAuthor> {
+  const payload = context.payload
+
+  const matches = (payload.head_commit?.message as string).match(/\(#(\d+)\)$/)
```

Comment:
Minor, but thoughts on leveraging the existing `isPushEvent` type predicate in favor of this type assertion, similar to what we do here?

https://github.com/Fieldguide/action-slack-deploy-pipeline/blob/fec653bf01c793a0c049e33280f0ab98e4bc1e5c/src/github/getSummaryMessage.ts#L120-L125

Tangentially related, but are there any other webhook events from which `github-merge-queue[bot]` would send?

Reply:
My understanding is at the moment it can only ever be a `push` event because the only use case for the merge queue is merging into the default branch. That may change in the future tho.

Reply:
In that case, would it make sense to be defensive, surfacing a clear warning/message if a merge user comes from a non-push context?

Reply:
Yea, set things up like:
```ts
if (isPushEvent(context)) {
  commit = context.payload.head_commit
} else {
  throw new Error(
    `Encountered Merge Queue Bot user in non push event: '${context.eventName}'.`
  )
}
```